### PR TITLE
Minimize Configuration broadcast/serialization overhead for HadoopRDDs/Hive-partition scans.

### PR DESCRIPTION
**Pull request by harveyfeng**
Previously, there was one broadcasted Configuration for each HadoopRDD created for a Hive partition that's stored on disk - e.g., for a table scan over 1000 partitions, there are 1000 Configuration broadcasts.
This change uses the HadoopFileRDD, so that there is one broadcasted Configuration shared across multiple HadoopRDDs created for the table scan.

Also, moved row deserialization from processPartition() to preprocessRDD(). This makes it more consistent with partitioned table reads/processing, and allows us to annotate HiveConf as transient, which should reduce Spark task sizes.

### CODE REVIEW

File: src/main/scala/shark/execution/TableScanOperator.scala [hunk 1]
```diff
@@ -214,15 +217,53 @@ class TableScanOperator extends TopOperator[HiveTableScanOperator] with HiveTopO
    * Create a RDD representing the table (with or without partitions).
    */
   override def preprocessRdd(rdd: RDD[_]): RDD[_] = {
+    // copyTableJobPropertiesToConf and pushFilters both take in JobConfs.
+    val localJobConf = new JobConf(localHconf)
```
Comment (rxin) [inline src/main/scala/shark/execution/TableScanOperator.scala]: Can we move the job conf setting into its own function?

**Comment by rxin** [comment]
Hi Harvey - this looks great. 

I had only one comment, and please make sure this builds with your latest change to Spark.

**Comment by harveyfeng** [comment]
TableScanOperator updated, PTAL

**Comment by harveyfeng** [comment]
Going to reopen a fresh PR tomorrow, once the Spark API changes are reflected.

**Pull request closed without merging**